### PR TITLE
Add the annotate gem and configure it

### PR DESCRIPTION
**Pull request by dkniffin**
### Why?

- #60 
- The annotations this gem provides removes the need to reference `schema.rb` so often, because the attributes and their types are listed at the top of the model files

### What changed?

- Added the annotate gem to the gemfile
- Added our `auto_annotate_models.rake` file, and code to copy it into place in the project

### CODE REVIEW

**Review by dacur — APPROVED** [review]
this LGTM 👍 

maybe i'm the only person that doesn't like this gem, but perhaps it should be optional?  if everyone else at sb seems to like it, then maybe it should just be included.  /shrug

**Comment by dkniffin** [comment]
@dacur Yeah. I'll ask in slack on Monday. I'm fine making it optional, but I know every project I come onto I've wanted it. I find it makes it easier to understand the model quickly if all the attributes are listed at the top. Otherwise, you have to look at two files to get all the information.

Is there any particular reason you don't like the gem?

**Comment by dacur** [comment]
cool, thanks for considering my input.  yeah, i remember one of the projects i was on there were so many attributes (comments) that i would have to scroll down to see the actual code in the model.  it's definitely just a personal preference.  i always use the schema as a reference anyway.  we're using this gem on roamd, so i'm trying it out again.  😄

**Comment by prawl** [comment]
When I'm looking at a model, I really like this gem and think it's faster to get context when the attributes are all listed on the top.  The only downside is when you make a migration and the annotated attributes are updated making it seem like PR's are bigger.  Just my two cents, but I'd like to keep this in.

**Comment by dacur** [comment]
example User model:

screen shot 2018-02-12 at 2 31 57 pm

File: templates/auto_annotate_models.rake [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      'ignore_columns'            => nil,
+      'ignore_routes'             => nil,
+      'ignore_unknown_models'     => 'false',
+      'hide_limit_column_types'   => 'integer,boolean',
+      'hide_default_column_types' => 'json,jsonb,hstore',
+      'skip_on_db_migrate'        => 'false',
+      'format_bare'               => 'true',
+      'format_rdoc'               => 'false',
+      'format_markdown'           => 'false',
+      'sort'                      => 'false',
```
Comment (TylerRockwell) [inline templates/auto_annotate_models.rake]: It might be nice to set this to true

**Comment by TylerRockwell** [comment]
This gem does have a couple downsides that have been mentioned above. That being said, I really enjoy not having to open up  `schema.rb` every time I want to check if a field exists or if I forgot the exact name of a field.

I don't care either way if it's optional, but I'd really like to have the gem be installed and configured (it's noisy as hell on default settings) by Boxcar.

**Comment by dkniffin** [comment]
It sounds like we're almost all agreed that we like annotate, and David has said it's fine to merge this, so I merged it. Thanks for your opinions everyone!

**Pull request merged**